### DISCUSSION TITLE
Feature request: Create enums from values in tables

### DISCUSSION THREAD

**Discussion by ajfriend** [duckdb/duckdb#5037] [General]
It would be nice if there were some syntax to allow users to create enums from string values found in tables. For example, maybe something like:

```sql
CREATE TYPE node_enum AS ENUM (select node_name from tbl_graph);
```

This would allow me to get much of the way towards the "Automatic Storage Encoding" idea described in this blog post: https://duckdb.org/2021/11/26/duck-enum.html#what-about-the-sequels. I would just be doing some of the work manually, rather than it happening invisibly to the user.

**Comment by ajfriend** [comment]
I can currently do something like this when working in a language like Python, but it requires me to extract the names from the table  into Python land, construct a string like `CREATE TYPE lotr_race AS ENUM ...` which could include many many unique strings, and send that command back to duckdb. It would be great if I could do all of that while staying within duckdb.

**Comment by hannes** [comment]
@pdet is there a way of creating an ENUM from a query result?

**Reply by pdet** [reply]
Not yet. But I do think we should eventually do that, creating enums 'manually' is a bit painful

**Reply by ajfriend** [reply]
Awesome! Any chance there would be something like an issue to keep track of this?

**Reply by pdet** [reply]
I don't think there is one yet. Please feel free to do so!

**Reply by ajfriend** [reply]
Cool. This started as an issue but got converted to a discussion. Is there a way to convert this back to an issue, or should I create a new one?

**Reply by ajfriend** [reply]
Created one here: https://github.com/duckdb/duckdb/issues/5106
